src/metabase/driver/sql/query_processor.clj

```diff
@@ -1399,7 +1404,8 @@
 (defn mbql->honeysql
   "Build the HoneySQL form we will compile to SQL and execute."
   [driver {inner-query :query}]
-  (binding [hx/*honey-sql-version* (honey-sql-version driver)]
+  (binding [driver/*driver*        driver
```

Comment:
Not strictly related but when writing a test I noticed it errored if `driver/*driver*` was unbound, which seemed ridiculous because `driver` is an explicit parameter to this function.